### PR TITLE
SNOW-1226614: Add parameter to disable SAML url check

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Add parameter to disable SAML url check

### Checklist
- [ ] Code compiles correctly
- [ ] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing (`dotnet test`)
- [ ] Extended the README / documentation, if necessary
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data/Core/Session/SFSessionProperty.cs [hunk 1]
```diff
@@ -110,7 +110,9 @@ internal enum SFSessionProperty
         [SFSessionPropertyAttr(required = false, defaultValue = "60m")]
         EXPIRATIONTIMEOUT,
         [SFSessionPropertyAttr(required = false, defaultValue = "true")]
-        POOLINGENABLED
+        POOLINGENABLED,
+        [SFSessionPropertyAttr(required = false, defaultValue = "false")]
+        DISABLE_SAML_URL_CHECK
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data/Core/Session/SFSessionProperty.cs:115]: Please add description to https://github.com/snowflakedb/snowflake-connector-net/blob/master/doc/Connecting.md

File: Snowflake.Data/Core/Authenticator/OktaAuthenticator.cs [hunk 2]
```diff
@@ -88,7 +88,10 @@ async Task IAuthenticator.AuthenticateAsync(CancellationToken cancellationToken)
                     _rawSamlTokenHtmlString = await samlRawResponse.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
 #endif
                     s_logger.Debug("step 5: Verify postback URL in SAML response");
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data/Core/Authenticator/OktaAuthenticator.cs:90]: Can we add debug info about skippig this step?

**Review by sfc-gh-dstempniak — CHANGES_REQUESTED** [review]

File: Snowflake.Data/Core/Session/SFSessionProperty.cs (hunk 1 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/Session/SFSessionProperty.cs:115]: Added description

File: Snowflake.Data/Core/Authenticator/OktaAuthenticator.cs (hunk 2 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/Authenticator/OktaAuthenticator.cs:90]: Sure, added a log when the check is skipped

**Review by sfc-gh-dstempniak — APPROVED** [review]

**Pull request merged**